src/wallet/walletdb.cpp

```diff
@@ -669,10 +688,23 @@ static DBErrors LoadLegacyWalletRecords(CWallet* pwallet, DatabaseBatch& batch,
                 chain.nExternalChainCounter = std::max(chain.nExternalChainCounter, index + 1);
             }
         }
+
+        pwallet->GetOrCreateLegacyScriptPubKeyMan()->LoadKeyMetadata(vchPubKey.GetID(), keyMeta);
+
         return DBErrors::LOAD_OK;
     });
     result = std::max(result, keymeta_res.m_result);
 
+    // Update any changed metadata
+    if (updated_metas.size()) {
+        LegacyScriptPubKeyMan* spkm = pwallet->GetLegacyScriptPubKeyMan();
+        Assert(spkm);
+        LOCK(spkm->cs_KeyStore);
+        for (const auto& pubkey : updated_metas) {
+            w_batch.WriteKeyMetadata(spkm->mapKeyMetadata.at(pubkey.GetID()), pubkey, true);
```

Comment:
nit: should check `WriteKeyMetadata` return value.

Reply:
Done